gnosis/safe/proxy_factory.py

```diff
@@ -57,6 +66,36 @@ def deploy_contract(
             deployer_account, constructor_data
         )
 
+    @cache
+    def get_proxy_creation_code(self) -> bytes:
+        """
+        :return: Creation code used for the Proxy deployment.
+            With this it is easily possible to calculate predicted address.
+        """
+        return self.contract.functions.proxyCreationCode().call()
+
+    @cache
+    def get_proxy_runtime_code(self) -> Optional[bytes]:
+        """
+        :return: Runtime code of a deployed Proxy. For v1.4.1 onwards the method is not available, so `None`
+            will be returned
+        """
+        if hasattr(self.contract.functions, "proxyRuntimeCode"):
+            return self.contract.functions.proxyRuntimeCode().call()
+
+    def get_deploy_function(self, chain_specific: bool) -> ContractFunction:
+        if chain_specific:
```

Comment:
If this feature is available only in 1.4.1, why not implement it in the 1.4.1 class?

Reply:
Good point. I was trying to make something compatible across all the ProxyFactories, but you are right